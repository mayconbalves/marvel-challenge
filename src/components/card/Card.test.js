import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import Card from './'

describe('<Card />', () => {
  const mockStore = configureStore([thunk])
  let store
  store = mockStore({
    charactersReducer: { characters: [] }
  })

  let props = {
    character: {
      thumbnail: {
        path: 'http://i.annihil.us/u/prod/marvel/i/mg/6/30/4c7c64437b5a1',
        extension: 'jpg'
      }
    },
    likes: [],
    handleFavorite: () => {},
    handleDesfavor: () => {}
  }

  it('should render correct empty img', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <Card {...props} />
        </Provider>
      </MemoryRouter>
    )

    expect(
      screen.getByTestId(/favoritar/i)
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
