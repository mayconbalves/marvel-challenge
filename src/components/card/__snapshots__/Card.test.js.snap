// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render correct empty img 1`] = `
<div
  class="sc-bdfBwQ gGfdFX"
>
  <a
    href="/characters/undefined"
  >
    <img
      class="sc-gsTCUz iHzmRF"
      src="http://i.annihil.us/u/prod/marvel/i/mg/6/30/4c7c64437b5a1.jpg"
    />
  </a>
  <div
    class="sc-dlfnbm inTuuj"
  >
    <p />
    <img
      alt="favoritar"
      class="sc-hKgILt jagTBZ"
      data-testid="favoritar"
      src="empty_heart.png"
    />
  </div>
</div>
`;
